var/spack/repos/builtin/packages/ascent/package.py

```diff
@@ -599,6 +630,41 @@ def hostconfig(self):
         else:
             cfg.write(cmake_cache_entry("ENABLE_OPENMP", "OFF"))
 
+        #######################
+        # HIP
+        #######################
+
+        cfg.write("# HIP Support\n")
+
+        if spec.satisfies("+rocm"):
+            cfg.write(cmake_cache_entry("ENABLE_HIP", "ON"))
+            cfg.write(cmake_cache_entry("KOKKOS_DIR", spec["kokkos"].prefix))
+
+            amdgpu_archs = ";".join(spec.variants["amdgpu_target"].value)
+            cfg.write(cmake_cache_path("HIP_ROOT_DIR", f"{spec['hip'].prefix}"))
+            cfg.write(cmake_cache_path("HIP_CLANG_PATH", f"{spec['llvm-amdgpu'].prefix.bin}"))
+            cfg.write(cmake_cache_string("CMAKE_HIP_ARCHITECTURES", amdgpu_archs))
+
+            clang_bindir = spec["llvm-amdgpu"].prefix.bin
+            cfg.write(cmake_cache_path("CMAKE_C_COMPILER", f"{clang_bindir}/clang", force=True))
+            cfg.write(
+                cmake_cache_path("CMAKE_CXX_COMPILER", f"{clang_bindir}/clang++", force=True)
+            )
+
+            # This is needed for Kokkos
+            cfg.write(cmake_cache_entry("CMAKE_CXX_STANDARD", "17"))
```

Comment:
I think there is a better way to do this...but this is the most straightforward.

```python
if spec["kokkos"].satisfies("cxxstd=11"):
  cxxstd="11"
elif spec["kokkos"].satisfies("cxxstd=14"):
  cxxstd="14"
elif spec["kokkos"].satisfies("cxxstd=17"):
  cxxstd="17"
elif spec["kokkos"].satisfies("cxxstd=20"):
  cxxstd="20"

cfg.write(cmake_cache_entry("CMAKE_CXX_STANDARD", cxxstd))
```

Reply:
That looks good, we should add:

```
cfg.write(cmake_cache_entry("BLT_CXX_STD", cxxstd))
```